app/docker-compose.yml

```diff
@@ -0,0 +1,30 @@
+version: '3'
+services:
+  flaskapp:
+    build: 
+      context: ./ui
+      dockerfile: Dockerfile-ui
+    ports:
+      - "5090:5090"
+    networks:
+      - web_nw
+  nginx:
+    image: "nginx:1.13.5"
+    ports:
+      - "8080:80"
+    volumes:
+      - ./nginx:/etc/nginx/conf.d
+    networks:
+      - web_nw
+    depends_on: 
+      - flaskapp
+  elasticsearch:
+    image: "docker.elastic.co/elasticsearch/elasticsearch:5.4.3"
+    environment:
+      - "discovery.type=single-node"
+      - "xpack.security.enabled=false"
+    ports:
+      - "9200:9200"
+networks:
```

Comment:
curious, more for my learning - what does this do?

Reply:
doing this tells `docker-compose` to network all the services together so they can talk to each other. Doing this, all services in the same network will be able to talk to each other over `<servicename>:<serviceport>`

so e.g. other services will be able to use `http://elasticseach:9200` in their code to hit the ES service